zerver/views/streams.py

```diff
@@ -384,19 +384,33 @@ def update_stream_backend(
     if stream_post_policy is not None:
         do_change_stream_post_policy(stream, stream_post_policy, acting_user=user_profile)
 
-    if can_remove_subscribers_group_id is not None:
-        if sub is None and stream.invite_only:
-            # Admins cannot change this setting for unsubscribed private streams.
-            raise JsonableError(_("Invalid stream ID"))
-
-        user_group = access_user_group_for_setting(
-            can_remove_subscribers_group_id,
-            user_profile,
-            setting_name="can_remove_subscribers_group",
-            require_system_group=True,
-        )
-
-        do_change_can_remove_subscribers_group(stream, user_group, acting_user=user_profile)
+    for setting_name, permissions_configuration in Stream.stream_permission_group_settings.items():
+        request_settings_dict = locals()
+        setting_group_id_name = setting_name + "_id"
+
+        if setting_group_id_name not in request_settings_dict:  # nocoverage
+            continue
+
+        if request_settings_dict[setting_group_id_name] is not None and request_settings_dict[
+            setting_group_id_name
+        ] != getattr(stream, setting_name):
+            if sub is None and stream.invite_only:
+                # Admins cannot change this setting for unsubscribed private streams.
+                raise JsonableError(_("Invalid stream ID"))
```

Comment:
We probably want to make this behavior controlled by a `GroupPermissionSetting` flag for whether this setting can be changed by unsubscribed administrators -- the idea being that the code here doesn't have special cases for particular settings. But that can be a follow-up.

Maybe `require_history_access_to_edit` would be a reasonable name for the flag?